### PR TITLE
feature/2 mechanism to set span kind depending on operation name

### PR DESCRIPTION
**Pull request by dkrizic**
- **Basic structure, but not implemented yet**
- **Passing the spankindSelector**
- **Test added**

### CODE REVIEW

**Comment by ravilushqa** [comment]
The code looks good to me!
I’ve merged a smaller PR, which caused a test conflict in this one. Could you please resolve the conflict and address a small linter issue? Once those are fixed, it’ll be ready for merge, and I’ll follow up with a new release.

Thanks!

**Comment by dkrizic** [comment]
Yes, the two PRs are actually touching the same code, but since I made them independent, a collision is expected. I will sync and update the PR. 

BTW: assert.Equal expects first "expected", then "actual". I changed this in the one place.

**Comment by dkrizic** [comment]
The conflict has been resolved.

**Comment by ravilushqa** [comment]
> The conflict has been resolved.

looks like something goes wrong an everything is failing now 😅

**Comment by dkrizic** [comment]
What do you mean? Can you fix the pipeline on main and I will update this branch.

**Comment by dkrizic** [comment]
I just see that the tests are failing on my side, I will update this

**Comment by dkrizic** [comment]
Tests are fix, the rest are your pipelines.

File: gqlgen.go [hunk 1]
```diff
@@ -182,6 +191,12 @@ func alwaysTrue() FieldsPredicateFunc {
 	}
 }
 
+func alwaysServer() SpanKindSelectorFunc {
+	return func(operationName string) oteltrace.SpanKind {
```
Comment (ravilushqa) [inline gqlgen.go:195]:
```suggestion
	return func(_ string) oteltrace.SpanKind {
```

**Comment by ravilushqa** [comment]
Thanks for your contribution!
I’ve taken care of the linter issue, and everything looks good to go now.

**Comment by dkrizic** [comment]
Grüsse aus Frankfurt nach Berlin.

**Comment by ravilushqa** [comment]
> Grüsse aus Frankfurt nach Berlin.

I'm new in Berlin so
Mein Deutsch ist nicht besonders gut,

**Pull request merged**